src/core/memblock.h

```diff
@@ -0,0 +1,17 @@
+#ifndef CORE_MEM_BLOCK_H
+#define CORE_MEM_BLOCK_H
+
+#include <stdint.h>
+
+typedef struct {
+	char *mem;
```

Comment:
Watch out for the proper indentation, it should be 4 spaces.

Also, use `memory`, not `mem`. Don't cut on words unnecessarily.